### PR TITLE
Add timer to critical sections

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/nrf_802154_critical_section.c [hunk 1]
```diff
@@ -125,14 +217,63 @@ static bool critical_section_enter(bool forced)
...
 
-    nrf_802154_log(EVENT_TRACE_EXIT, FUNCTION_CRIT_SECT_ENTER);
     return result;
 }
 
+static void critical_section_exit(void)
+{
+    uint8_t     cnt;
+    static bool exiting_crit_sect;
+    rsch_evt_t  rsch_evt= rsch_pending_evt_clear();
```
Comment (pszkotak) [inline src/nrf_802154_critical_section.c]: nit: space missing

File: src/nrf_802154_critical_section.c [hunk 2]
```diff
@@ -125,14 +217,63 @@ static bool critical_section_enter(bool forced)
...
     while (__STREXB(cnt + 1, &m_nested_critical_section_counter));
 
-    nrf_802154_log(EVENT_TRACE_EXIT, FUNCTION_CRIT_SECT_ENTER);
     return result;
 }
 
+static void critical_section_exit(void)
+{
+    uint8_t     cnt;
+    static bool exiting_crit_sect;
```
Comment (LuDuda) [inline src/nrf_802154_critical_section.c]: nit: Maybe we can move it to the top of the file?

File: src/nrf_802154_swi.h [hunk 3]
```diff
@@ -164,6 +164,9 @@ void nrf_802154_swi_receive(nrf_802154_term_t              term_lvl,
  * @param[in]   req_orig         Module that originates this request.
  * @param[in]   p_data           Pointer to PSDU of the frame to transmit.
  * @param[in]   cca              If the driver should perform CCA procedure before transmission.
+ * @param[in]  immediate         If true, the driver schedules transmission immediately or never;
```
Comment (LuDuda) [inline src/nrf_802154_swi.h]: nit: alignment

File: src/nrf_802154_critical_section.c (hunk 2 above)
Comment (hubertmis) [reply src/nrf_802154_critical_section.c]: I don't think so. It is local static variable. It should be encapsulated in a function. There is no need to extend its scope.

**Pull request merged**